core/src/main/java/com/netflix/conductor/core/execution/tasks/Join.java

```diff
@@ -60,7 +60,10 @@ public boolean execute(
             if (hasFailures) {
                 failureReason.append(forkedTask.getReasonForIncompletion()).append(" ");
             }
-            task.addOutput(joinOnRef, forkedTask.getOutputData());
+            // Only add to task output if it's not empty
+            if (!forkedTask.getOutputData().isEmpty()) {
```

Comment:
Does `getOutputData()` always return a non-null value?

Reply:
yes, we always init it to empty map

Reply:
FYI this breaks backward compatibility if anyone relied on this to be present.